src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -405,6 +419,11 @@ public void unloadCube(int cubeX, int cubeY, int cubeZ) {
 		if (cubeIsNearSpawn(cubeX, cubeY, cubeZ)) {
 			return;
 		}
+		
+		// Do not unload cubes which are required for generating currently loaded cubes.
+		if (dependencyManager.isRequired(cubeX, cubeY, cubeZ)) {
+			return;
```

Comment:
I will test it myself soon. I think I had a piece of code for 2d visual representation of ServerCubeCache somewhere...